.github/workflows/elixir-build-check.yaml

```diff
@@ -86,3 +86,22 @@ jobs:
           mix-env: ${{ inputs.mix-env }}
       - name: Check for circular dependencies
         run: mix xref graph --label compile-connected --fail-above 0
+  format:
+    name: Validate formatting complies with `mix format`
+    needs: [dependencies]
+    runs-on: ${{ inputs.runs-on }}
+    steps:
+      - name: Fetch the full repository so we can perform a Git diff
+        uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Fetch dependencies as they may be needed by the formatter
+        uses: rentpath/actions-services-elixir-deps@v1
+        with:
+          elixir-version: ${{ inputs.elixir-version }}
+          erlang-version: ${{ inputs.erlang-version }}
+          mix-env: ${{ inputs.mix-env }}
+      - name: Check formatting for Elixir files that have changed
+        run: |
+          ref="origin/${{ github.event.repository.default_branch }}"
+          git diff --name-only "$ref" | egrep '(\.ex|\.exs)+$' | xargs mix format --check-formatted --dry-run -
```

Comment:
I struggled a bit getting this git diff command to work. `${{ github.event.repository.default_branch }}` seems like the best way of getting the default branch in Github Actions. We can't hardcode `main` or `master` here because the default branch varies by service.